### ISSUE TITLE
Record audio in background

### ISSUE THREAD

**Issue by juligupta** [hyochan/react-native-audio-recorder-player#323]
> Please fill the template to help you out. Also, please try the `Example` project compare before submiting the issue when you have certain issue with your project setup.

### Version of react-native-audio-recorder-player
2.5.3

### Platforms you faced the error (IOS or Android or both?)
Android
### Expected behavior
When application goes to background, the audio recording should continue

### Actual behavior
When application goes to background, the audio recording stops

**Comment by hyochan** [comment]
Since version 2+ is not maintained, could you also try our recent version which currently is `3.0.8`?

**Comment by CodeFromAnywhere** [comment]
I would really like to know if anyone got this to work. I want to record audio 24/7 and stream the recording to an api without having the app open (personal research project)

**Comment by GowthamOutlook** [comment]
iOS working fine I have issues in android alone in background state audio is not recording in Android @CodeFromAnywhere

**Comment by MacDinhThanh** [comment]
the same issue

**Comment by the-mad** [comment]
Any news on this ?

**Comment by the-mad** [comment]
Ok, if someone still has this issue, I figured out for iOS that you simply have to add : 
```
<key>UIBackgroundModes</key>
<array>
    <string>audio</string>
</array>
```
In your `Info.plist` file. By doing this, the recording should continue if the app goes in background.

**Comment by askpatrickw** [comment]
I'm wondering, is there any update on this for Android?

If someone can provide guidance, my team is willing to help add this feature.

**Comment by lukailun** [comment]
> I'm wondering, is there any update on this for Android?
> 
> If someone can provide guidance, my team is willing to help add this feature.

Include an instance of MediaRecorder in a foreground service should fix this.

```
Note: On devices running Android 9 (API level 28) or higher, apps running in the background cannot access the microphone. Therefore, your app should record audio only when it's in the foreground or when you include an instance of MediaRecorder in a [foreground service](https://developer.android.com/guide/components/services#Foreground).
```

https://developer.android.com/media/platform/mediarecorder

**Comment by anuja33n** [comment]
I am also facing the same issue. When user goes to background on android then recording has not been captured in background mode. Does the lib provide the support for this feature. Can you please confirm?

**Comment by anujapisemb** [comment]
I am also facing the same issue. When user goes to background on android then its doesn't captured recording in background mode. Does this lib provide the support for this feature. Can anyone please confirm?

**Comment by anujapisemb** [comment]
> > To fix background recording on Android, you can refer to this commit: https://github.com/hongdtt-ominext/react-native-nitro-sound/tree/fix-buf-call
> > Note: On iOS, you must still enable UIBackgroundModes.
> 
> Can we have a PR of this ?

Does this PR which resolves issue for android has merged?

**Comment by GabeConsalter** [comment]
> > > To fix background recording on Android, you can refer to this commit: https://github.com/hongdtt-ominext/react-native-nitro-sound/tree/fix-buf-call
> > > Note: On iOS, you must still enable UIBackgroundModes.
> > 
> > 
> > Can we have a PR of this ?
> 
> Is this working [@GabeConsalter](https://github.com/GabeConsalter) for background recording on android?

Actually don't work to me.

**Comment by hongdtt-ominext** [comment]
@anujapisemb yup. It works. Have you checked the AndroidManifest file to see these two lines? android.permission.FOREGROUND_SERVICE
android.permission.FOREGROUND_SERVICE_MICROPHONE
and have you check the folder android lib react-native-nitro-sound to see RecordingForegroundService.kt?

**Comment by hyochan** [comment]
This issue was filed against the pre-Nitro `react-native-audio-recorder-player` API. This repo has since been completely rewritten on top of [react-native-nitro-modules](https://github.com/mrousavy/nitro) as **react-native-nitro-sound**.

If this still reproduces on the latest `react-native-nitro-sound`, please open a fresh issue with:
- Library version (`react-native-nitro-sound`)
- React Native version
- Platform (iOS / Android)
- Minimal repro steps and logs

Closing as `stale-legacy` for now.

> 🤖 AI-native maintenance. See #787 for context.

**Comment by hyochan** [comment]
Closing as part of a stale-legacy sweep.

This issue was filed against the pre-Nitro `react-native-audio-recorder-player` codebase, hasn't seen activity in a long time, and the underlying API has since been rewritten on top of [Nitro Modules](https://github.com/mrousavy/nitro). Most legacy code paths referenced here no longer exist.

If this still affects you on **`react-native-nitro-sound` v3+**, please:
1. Open a fresh issue with a minimal repro on the current version, and
2. Include your iOS/Android version, RN version, and `nitro-modules` version.

Thanks for the original report — closing to keep the queue focused on actionable, current-version reports. 🙏

> 🤖 Auto-closed by daily triage.

**Comment by hyochan** [comment]
Closing as part of a stale-legacy sweep.

This issue was filed against the pre-Nitro `react-native-audio-recorder-player` codebase, hasn't seen activity in a long time, and the underlying API has since been rewritten on top of [Nitro Modules](https://github.com/mrousavy/nitro). Most legacy code paths referenced here no longer exist.

If this still affects you on **`react-native-nitro-sound` v3+**, please:
1. Open a fresh issue with a minimal repro on the current version, and
2. Include your iOS/Android version, RN version, and `nitro-modules` version.

Thanks for the original report — closing to keep the queue focused on actionable, current-version reports. 🙏

> 🤖 Auto-closed by daily triage.
